torch/csrc/mps/Module.cpp

```diff
@@ -74,7 +74,7 @@ static PyObject* MPSModule_isMacOS13orNewer(PyObject* _unused, PyObject* args) {
 
 static PyObject* MPSModule_synchronize(PyObject* _unused, PyObject* noargs) {
   HANDLE_TH_ERRORS
-  at::detail::getMPSHooks().deviceSynchronize();
+  at::detail::getMPSHooks().synchronizeStream();
```

Comment:
huh? 
That doesn't seem to match the current doc for this function? https://pytorch.org/docs/stable/generated/torch.mps.synchronize.html?highlight=synchronize#torch.mps.synchronize

Reply:
We'll have `synchronizeEvent()` API coming in a separate PR, so I wanted to make the distinction with `synchronizeStream()`. We want Python API to remain as `torch.mps.synchronize()` to have the CUDA equivalent, and don't think it matters what our internal MPSHooks implementation calls it. 

I'll rename MPSModule_synchronize() to MPSModule_synchronizeStream() , though.

Reply:
The internal name doesn't matter much but it makes things easier to follow if they have the same name?
To me `deviceSynchronize` sounds like you synchronize the whole device (all streams there) while `synchronizeStream` is synchronizing a single stream (the current one here I guess).

Reply:
Renamed it back to `deviceSynchronize`.